src/models/contact/index.js

```diff
@@ -58,9 +56,7 @@ export async function getContactAll(addressBook, types, fetch) {
     types.map(async (type) => {
       const dataset = await getContactIndexDataset(addressBook, type, fetch);
       return getThingAll(dataset)
-        .filter((contact) =>
-          getUrlAll(contact, rdf.type).includes(type.contactTypeUrl)
-        )
+        .filter((contact) => type.isOfType(contact))
```

Comment:
This needs to be reflected in the test. The current tests are flawed in that the mocked ContactPerson is also adding type `vcard:Individual`, which is wrong. We need to update the mocks to be closer to reality, and update the tests accordingly.